### PR TITLE
Added acceptation of flirts for ticket 106

### PR DESCRIPTION
**Pull request by HUSKUT**

### CODE REVIEW

File: lib/acceptFlirt.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import backendFetch from './backendFetch';
+
+export default async function acceptFlirt(flirtId: number): Promise<void> {
+  console.log('flirtid', flirtId);
```
Comment (SophiaH67) [inline lib/acceptFlirt.ts]: dwayne-johnson-looking
```suggestion
```

File: screens/MatchingScreen.tsx [hunk 2]
```diff
@@ -53,11 +54,14 @@ export default function MatchingScreen({
     getFlirtsForCompany()
       .then(setFlirts)
       .catch(() => {});
-    console.log('flirts', flirts);
   }, [refresh]);
 
+  let uniqueFlirts = flirts.filter((flirt, index, self) => {
+    return self.findIndex(f => f.student.name === flirt.student.name) === index;
+  });
+  // Prevents from showing duplicate flirts
```
Comment (SophiaH67) [inline screens/MatchingScreen.tsx]: Wait what, why are duplicate flirts allowed in the first place??? That sounds like a pretty significant bug in the backend...

**Review by SophiaH67 — CHANGES_REQUESTED** [review]

File: screens/MatchingScreen.tsx (hunk 2 above)
Comment (SophiaH67) [inline screens/MatchingScreen.tsx]:
```suggestion
```

File: screens/MatchingScreen.tsx [hunk 3]
```diff
@@ -71,16 +75,25 @@ export default function MatchingScreen({
...
         <View style={styles.main}>
           <TopBar ScreenName='Logout' Press={CustomLogout} />
           <ScrollView style={styles.content}>
-            {flirts.map(flirt => (
-              <Notification
-                title={flirt.student.name}
-                key={flirt.id}
-                message={flirt.student.profileSettings.description}
-              />
+            {uniqueFlirts.map(flirt => (
```
Comment (SophiaH67) [inline screens/MatchingScreen.tsx]:
```suggestion
            {flirts.map(flirt => (
```

File: screens/MatchingScreen.tsx [hunk 4]
```diff
@@ -71,16 +70,25 @@ export default function MatchingScreen({
...
+                  message={flirt.student.profileSettings.description}
+                  buttonMessage={'Accept Flirt'}
+                  action={() =>
+                    acceptFlirt(flirt.id)
+                      .then(getFlirtsForCompany)
+                      .then(setFlirts)
+                  }
+                />
+              </>
             ))}
```
Comment (SilGosker) [inline screens/MatchingScreen.tsx]:
```suggestion
                <Notification
                  title={flirt.student.name}
                  key={flirt.id}
                  message={flirt.student.profileSettings.description}
                  buttonMessage={'Accept Flirt'}
                  action={() =>
                    acceptFlirt(flirt.id)
                      .then(getFlirtsForCompany)
                      .then(setFlirts)
                  }
                />
            ))}
```

**Review by SilGosker — CHANGES_REQUESTED** [review]

**Review by IngmarDeRuiter — APPROVED** [review]

**Review by SilGosker — APPROVED** [review]

**Review by SophiaH67 — APPROVED** [review]

**Pull request merged**